homework_3/src/main/java/com/epam/spring/service/repository/impl/CategoryRepositoryImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.epam.spring.service.repository.impl;
+
+import com.epam.spring.exception.CategoryNotFoundException;
+import com.epam.spring.exception.ReceiptNotFoundException;
+import com.epam.spring.service.model.Category;
+import com.epam.spring.service.repository.CategoryRepository;
+import org.springframework.stereotype.Component;
+
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class CategoryRepositoryImpl implements CategoryRepository {
+
+    private final List<Category> categoryList = new ArrayList<>();
+
+    @Override
+    public Category getCategory(String name) {
+        Optional<Category> optional = categoryList.stream()
+                .filter(p -> p.getName().equals(name))
+                .findFirst();
+        if (!optional.isPresent()) {
+            throw new CategoryNotFoundException();
+        }
+        return optional.get();
+    }
+
+    @Override
+    public List<Category> listCategories() {
+        return new ArrayList<>(categoryList);
+    }
+
+    @Override
+    public Category createCategory(Category category) {
+        category.setId(categoryList.size() + 1);
+        category.setModifiedDate(Timestamp.valueOf(LocalDateTime.now()));
+        categoryList.add(category);
+        return category;
+    }
+
+    @Override
+    public Category updateCategory(int id, Category category) {
+        Optional<Category> optional = categoryList.stream().filter(p -> p.getId() == id).findFirst();
+        if (!optional.isPresent()) {
```

Comment:
Hi. Lets try to rewrite the code (46-51 rows) into more functional style using the following template
```
Category oldCategory = categoryList.stream()
        .filter([USING PREDICAT]).findFirst()
        .orElseThrow([THROWING CORRESPONDING EXCEPTION]);
```

Please perform this for all Repositories.
Thank you!